Sources/Core/OpaquePointer.swift

```diff
@@ -0,0 +1,48 @@
+// Copyright © 2022 Max Desiatov <max@desiatov.com>.
+// All Rights Reserved.
+// SPDX-License-Identifier: BSD-3
+
+@frozen
+public struct OpaquePointer {
+  @usableFromInline
+  internal var _rawValue: Builtin.RawPointer
+
+  @usableFromInline @_transparent
+  internal init(_ v: Builtin.RawPointer) {
+    _rawValue = v
+  }
+
+  @_transparent
+  public init?(bitPattern: Int) {
+    guard bitPattern != 0 else { return nil }
```

Comment:
Is there a reason to use `guard` here rather than `if bitPattern == 0 { return nil }`?